kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaHistoryManager.java

```diff
@@ -162,6 +167,39 @@ public static long getEarliestRecreatableCommit(Engine engine, Path logPath)
         }
     }
 
+    /** Get the persisted commit info (if available) for the given delta file. */
+    public static Optional<CommitInfo> getCommitInfoOpt(
```

Comment:
This is probably not the right place for this function. I would suggest moving it to `CommitInfo.java` for now and call it `readCommitInfoOptFromDelta`. @vkorukanti probably has a better idea about where this should be kept.